smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/KafkaRecordTest.java

```diff
@@ -41,6 +47,24 @@ public void testCreationOfMessageWithKeyAndValue() {
         assertThat(metadata.getTopic()).isNull();
         assertThat(metadata.getTimestamp()).isNull();
         assertThat(metadata.getHeaders()).isNull();
+        // TODO - we don't have duplicate metadata when the Message API is used directly
+        // KafkaTestBase.tempCompareLegacyAndApiMetadata(metadata, message);
```

Comment:
This is a slight hole in the current implementation. But I hope that it will not be problematic.

My thinking is that when people set this metadata using one key they can be expected to look it up with the same key. As I understand it this will only be a problem before the message is sent? 

Once KafkaSink receives it, it will make the metadata available under both keys.

Reply:
I agree, not a problem.